cmake/xyz_add_test.cmake

```diff
@@ -101,7 +101,7 @@ function(xyz_add_test)
 
         include(GoogleTest)
         gtest_discover_tests(${XYZ_NAME}
-            WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
+            WORKING_DIRECTORY $<TARGET_FILE_DIR:${XYZ_NAME}>
```

Comment:
Actually, I think the correct fix would be to ensure the runtime output-dir is set per target as such:

```    set_target_properties(${XYZ_NAME}
        PROPERTIES
            ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
            LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
            RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin
    )
```

But if this works also happy with that for now

Reply:
Happy to apply your fix but it does not seem to get rid of the error:

Is CMAKE_RUNTIME_OUTPUT_DIRECTORY the same as RUNTIME_OUTPUT_DIRECTORY?

Reply:
Ok, that make sense as `RUNTIME_OUTPUT_DIRECTORY` is limited in scope to just the target where as `CMAKE_RUNTIME_OUTPUT_DIRECTORY` is global in scope.  I think used in conjunction with your prior change will give us what we want, i.e. executables are in `build/bin`, without this they will be scattered throughout the build directory mirroring their offset from the project base to subdirectory.